server/src/test/java/com/vaadin/data/BeanBinderTest.java

```diff
@@ -545,4 +553,10 @@ private void assertInvalid(HasValue<?> field, String message) {
         assertSame(field, errors.get(0).getField());
         assertEquals(message, errors.get(0).getMessage().get());
     }
+
+    @Test
+    public void interface_extension_with_overwritten_property() {
+        Binder<InterfaceWithOverwrittenMethod> binder =
+                new Binder<>(InterfaceWithOverwrittenMethod.class);
```

Comment:
Can you please create a very simple test that the binding works? You know, bind the "property" property in binder to IntegerField or similar, then call readBean with some implementor of InterfaceWithOverwrittenMethod, then asserting that the IntegerField has desired value.